redis_cache.go

```diff
@@ -0,0 +1,138 @@
+package lcw
+
+import (
+	"sync/atomic"
+	"time"
+
+	redis "github.com/go-redis/redis/v7"
+	"github.com/pkg/errors"
+)
+
+// RedisValueSizeLimit is maximum allowed value size in Redis
+const RedisValueSizeLimit = 512 * 1024 * 1024
+
+// RedisCache implements LoadingCache for Redis.
+type RedisCache struct {
+	options
+	CacheStat
+	backend *redis.Client
+}
+
+// NewRedisCache makes Redis LoadingCache implementation.
+func NewRedisCache(backend *redis.Client, opts ...Option) (*RedisCache, error) {
+
+	res := RedisCache{
+		options: options{
+			ttl: 5 * time.Minute,
+		},
+	}
+	for _, opt := range opts {
+		if err := opt(&res.options); err != nil {
+			return nil, errors.Wrap(err, "failed to set cache option")
+		}
+	}
+
+	if res.maxValueSize <= 0 {
```

Comment:
i think the second part of the check is missing

`res.maxValueSize > RedisValueSizeLimit`